packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -403,23 +398,56 @@ export function extractParticipation(
 }
 
 /**
- * The state transition accepts incorrect target and head attestations.
- * We only need to validate the source checkpoint.
+ * Do those validations:
+ *   - Validate the source checkpoint
+ *   - Since we validated attestation's signature in gossip validation function,
+ *     we only need to validate the shuffling of attestation
+ *     is compatible to this state.
+ *     (see https://github.com/ChainSafe/lodestar/issues/4333)
  * @returns
  */
 export function isValidAttestationData(
-  currentEpoch: Epoch,
-  previousJustifiedCheckpoint: phase0.Checkpoint,
-  currentJustifiedCheckpoint: phase0.Checkpoint,
+  forkChoice: IForkChoice,
+  state: CachedBeaconStateAllForks,
   data: phase0.AttestationData
 ): boolean {
+  const {previousJustifiedCheckpoint, currentJustifiedCheckpoint} = state;
   let justifiedCheckpoint;
-  if (data.target.epoch === currentEpoch) {
+  const stateEpoch = state.epochCtx.epoch;
+  const targetEpoch = data.target.epoch;
+
+  if (targetEpoch === stateEpoch) {
     justifiedCheckpoint = currentJustifiedCheckpoint;
-  } else {
+  } else if (targetEpoch === stateEpoch - 1) {
     justifiedCheckpoint = previousJustifiedCheckpoint;
+  } else {
+    return false;
   }
-  return ssz.phase0.Checkpoint.equals(data.source, justifiedCheckpoint);
+
+  if (!ssz.phase0.Checkpoint.equals(data.source, justifiedCheckpoint)) return false;
+
+  // Shuffling can't have changed if we're in the first few epochs
+  if (stateEpoch < 2) {
+    return true;
+  }
+  // Otherwise the shuffling is determined by the block at the end of the target epoch
+  // minus the shuffling lookahead (usually 2). We call this the "pivot".
+  const pivotSlot = computeStartSlotAtEpoch(targetEpoch - 1) - 1;
+  const statePivotBlockRoot = toHexString(getBlockRootAtSlot(state, pivotSlot));
+
+  // Use fork choice's view of the block DAG to quickly evaluate whether the attestation's
+  // pivot block is the same as the current state's pivot block. If it is, then the
+  // attestation's shuffling is the same as the current state's.
+  // To account for skipped slots, find the first block at *or before* the pivot slot.
+  let pivotBlockRoot = "";
+  for (const protoBlock of forkChoice.iterateAncestorBlocks(toHexString(data.beaconBlockRoot))) {
```

Comment:
This can be optimized using the targetRoot of each protoBlock. So it always takes 2 hops at most:
- Get target of beaconBlockRoot.
- If target protoBlock > pivotSlot, get target's parent. Else return target